dds/DCPS/XTypes/IdlScanner.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include <DCPS/DdsDcps_pch.h>
+
+#ifndef OPENDDS_SAFETY_PROFILE
+#  include "IdlScanner.h"
+
+# include "DynamicTypeImpl.h"
+
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+namespace OpenDDS {
+namespace XTypes {
+
+IdlToken
+IdlScanner::scan_token(DDS::DynamicType_ptr type)
+{
+  OPENDDS_ASSERT(type);
+
+  DDS::DynamicType_var bt = get_base_type(type);
+
+  switch (bt->get_kind()) {
+  case TK_BOOLEAN: {
+    const IdlToken token = scan_identifier();
+    if (token.is_boolean()) {
+      return token;
+    }
+    break;
+  }
+  case TK_BYTE: {
+    const IdlToken token = scan_numeric_literal();
+    if (token.is_integer() && !token.is_signed() && token.integer_value() <= 255) {
+      return token;
+    }
+    break;
+  }
+  case TK_INT16: {
+    const IdlToken token = scan_numeric_literal();
+    const uint64_t limit = token.is_signed() ? 32768 : 32767;
+    if (token.is_integer() && token.integer_value() <= limit) {
+      return token;
+    }
+    break;
+  }
+  case TK_INT32: {
+    const IdlToken token = scan_numeric_literal();
+    const uint64_t limit = token.is_signed() ? 2147483648 : 2147483647;
+    if (token.is_integer() && token.integer_value() <= limit) {
+      return token;
+    }
+    break;
+  }
+  case TK_INT64: {
+    const IdlToken token = scan_numeric_literal();
+    const uint64_t limit = token.is_signed() ? 9223372036854775808UL : 9223372036854775807;
```

Comment:
`uint64_t` isn't C++03. Also use either C, C++, or ACE macros/functions for the limits?

Reply:
Or for this specifically I guess it looks like you're trying to stuff signed into unsigned. That seems like it might be problematic. Why not a separate signed field/type?

Reply:
Getting the limits from a macro is a good idea.  C/C++/ACE is going to give me something that's for the compiler and not IDL.  I looked at ACE and TAO for limit for IDL/CDR but didn't find anything.  Let me know if I missed something.